lib/cucumber/filters/retry.rb

```diff
@@ -0,0 +1,19 @@
+require 'cucumber/core/filter'
+require 'cucumber/running_test_case'
+require 'cucumber/events/bus'
+require 'cucumber/events/after_test_case'
+
+module Cucumber
+  module Filters
+    class Retry < Core::Filter.new(:configuration)
+
+      def test_case(test_case)
+        super 
+
+        configuration.on_event(:after_test_case) do |event|
```

Comment:
I'd have thought you'd need to register the event listener earlier than this, like maybe in the filter's constructor. Does it work?